prow/crier/reporters/gerrit/reporter.go

```diff
@@ -411,7 +413,22 @@ func (c *Client) Report(ctx context.Context, logger *logrus.Entry, pj *v1.ProwJo
 		// Retry without voting on a label
 		message := fmt.Sprintf("[NOTICE]: Prow Bot cannot access %s label!\n%s", reportLabel, message)
 		if err := c.gc.SetReview(gerritInstance, gerritID, gerritRevision, message, nil); err != nil {
-			logger.WithError(err).WithField("gerrit_id", gerritID).Errorf("Failed to set plain review on change ID.")
+			if change == nil {
+				var debugErr error
+				change, debugErr = c.gc.GetChange(gerritInstance, gerritID)
+				if debugErr != nil {
+					logger.WithError(debugErr).WithField("gerrit_id", gerritID).Errorf("Getting change failed. This is trying to help determine why SetReview failed.")
+				}
+			}
+			if change != nil {
+				// keys of `Revisions` are the revision strings, see
+				// https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#change-info
+				if _, ok := change.Revisions[gerritRevision]; !ok {
+					logger.WithField("gerrit_id", gerritID).Debug("The revision to be commented is missing, swallow error.")
+					return nil, nil, nil
```

Comment:
Wouldn't it be better to let the rest of the function finish so that the previous report state is updated? Otherwise all other jobs for the ref will face the same problem. This isn't an issue just a bit unnecessary.

Reply:
good point, done